File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_set<int>st;
        for(int i : arr)
        {
            if(st.find(i*2) != st.end()) return true;
            if(i%2 == 0 && st.find(i/2) != st.end()) return true;
            st.insert(i);
        }
        return false;
    }
};

File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashSet<Integer>st = new HashSet<>();
        for(int i : arr)
        {
            if(st.contains(i*2)) return true;
            if(i%2 == 0 && st.contains(i/2)) return true;
            st.add(i);
        }
        return false;
    }
}